src/ts/shared/errors/DbServerError.ts

```diff
@@ -1,27 +1,31 @@
 import { dbErrors } from '#shared/constants';
 import BaseError from './BaseError';
 
-interface ConstructorOptions {
-	message?: string;
+interface DbServerErrorConstructorOptionsMessage {
+	message: string;
+}
+
+interface DbServerErrorConstructorOptionsCode {
 	errorCode: number | string;
 }
 
+type DbServerErrorConstructorOptions =
+	| DbServerErrorConstructorOptionsMessage
+	| DbServerErrorConstructorOptionsCode;
+
 /**
  * Error thrown when an error is encountered on the database server
  */
 class DbServerError extends BaseError {
-	public constructor({ message: overrideMessage, errorCode }: ConstructorOptions) {
+	public constructor(options: DbServerErrorConstructorOptions) {
```

Comment:
There were two different kinds of calls into this error.  One with just a message, and another with a code provided.  I had to change the definition to support both.